### PR TITLE
Emit 'unsubsribe' event when client has clean=true on disconnect #145

### PR DESCRIPTION
**Pull request by cordovapolymer**
@mcollina , i've implemented #145, please review if it's ok to merge.

### CODE REVIEW

File: lib/handlers/unsubscribe.js [hunk 1]
```diff
@@ -68,7 +68,7 @@ function completeUnsubscribe (err) {
     return client.emit('error', err)
   }
 
-  if (!packet.close && packet.unsubscriptions.length > 0) {
+  if (client.clean === true && packet.unsubscriptions.length > 0) {
```
Comment (mcollina) [inline lib/handlers/unsubscribe.js]: the check for `packet.close`  must stay. See https://github.com/mcollina/aedes/blob/275bda32461ea8053c8610b8a3e5b6f0f838f1b6/lib/client.js#L214.
Basically if `packet.close === true && client.clean === false` we should __not__ emit.

Comment (cordovapolymer) [reply lib/handlers/unsubscribe.js]: @mcollina , thank you for a clarification, please see the new commit

File: test/meta.js [hunk 2]
```diff
@@ -123,7 +123,7 @@ test('emit unsubscribe event', function (t) {
   t.plan(6)
 
   var broker = aedes()
-  var s = connect(setup(broker), { clientId: 'abcde' })
+  var s = connect(setup(broker), { clean: true, clientId: 'abcde' })
```
Comment (mcollina) [inline test/meta.js]: can you add another test that fails if the event is emitted when clean is false and the connection ends abruptly?

Comment (mcollina) [reply test/meta.js]: foget about it, it's already added.

**Comment by mcollina** [comment]
LGTM

**Pull request merged**